stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -1266,10 +1266,10 @@ function pinv(A::StridedMatrix{T}, rtol::Real) where T
     Sinv        = zeros(Stype, length(SVD.S))
     index       = SVD.S .> rtol*maximum(SVD.S)
     Sinv[index] = one(Stype) ./ SVD.S[index]
-    Sinv[findall(.!isfinite.(Sinv))] .= zero(Stype)
```

Comment:
Why is this change needed?

Reply:
It's not needed for this change. However, `zero(Stype)` isn't needed since the array is already allocated with `Stype` elements and `0` is easier to read than `zero(Stype)`.

Reply:
Aren't there places where `zero(T)` works but `convert(T, 0)` fails?  I'd let sleeping dogs lie here, especially since we also use `one`  and `zeros` immediately surrounding this line.

Reply:
There are but they don't apply here since `Stype` is the type the singular values. It's mainly for non-number types like arrays that you can't rely on converting from `0`. This change is unrelated to the PR so I've removed it.

Reply:
That is true. The singular values have units so that would fail. Unfortunately, it would be a bit complicated to make the svd solver in `GenericLinearAlgebra` work for `Unitful.Quantity` because it's only a subtype of `Number`.